src/Changelog/ReleaseChangelogViaKeepAChangelog.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Changelog;
+
+use Laminas\AutomaticReleases\Git\CommitFile;
+use Laminas\AutomaticReleases\Git\Push;
+use Laminas\AutomaticReleases\Git\Value\BranchName;
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Phly\KeepAChangelog\Common\DiscoverChangelogEntryListener;
+use Phly\KeepAChangelog\Config;
+use Phly\KeepAChangelog\Version\ReadyLatestChangelogEvent;
+use Phly\KeepAChangelog\Version\SetDateForChangelogReleaseListener;
+use Symfony\Component\Console\Output\NullOutput;
+
+use function date;
+use function file_exists;
+use function sprintf;
+
+final class ReleaseChangelogViaKeepAChangelog implements ReleaseChangelog
+{
+    private CommitFile $commitFile;
+    private Push $push;
+
+    public function __construct(
+        CommitFile $commitFile,
+        Push $push
+    ) {
+        $this->commitFile = $commitFile;
+        $this->push       = $push;
+    }
+
+    /**
+     * @psalm-param non-empty-string $repositoryDirectory
+     */
+    public function __invoke(
+        string $repositoryDirectory,
+        SemVerVersion $version,
+        BranchName $sourceBranch
+    ): void {
+        $changelogFile = $repositoryDirectory . '/CHANGELOG.md';
+        if (! file_exists($changelogFile)) {
+            // No changelog
+            return;
+        }
+
+        $versionString = $version->fullReleaseName();
+
+        if (! $this->updateChangelog($changelogFile, $versionString)) {
+            // Failure to update; nothing to commit
+            return;
+        }
+
+        ($this->commitFile)(
+            $repositoryDirectory,
+            $sourceBranch,
+            'CHANGELOG.md',
+            sprintf('%s readiness', $versionString)
```

Comment:
`1.2.3 readiness` doesn't sound like an awesome commit text: can we use something that explains that we did just provide a changelog text?

Reply:
Will do!